templates/common/_base/units/machine-config-daemon-pull.service.yaml

```diff
@@ -16,7 +16,7 @@ contents: |
   RemainAfterExit=yes
   # See https://github.com/coreos/fedora-coreos-tracker/issues/354
   ExecStart=/bin/sh -c '/bin/mkdir -p /run/bin && chcon --reference=/usr/bin /run/bin'
-  ExecStart=/bin/sh -c "/usr/bin/podman pull --authfile=/var/lib/kubelet/config.json --quiet '{{ .Images.gcpRoutesControllerKey }}'"
+  ExecStart=/bin/sh -c "while ! /usr/bin/podman pull --authfile=/var/lib/kubelet/config.json --quiet '{{ .Images.gcpRoutesControllerKey }}'; do sleep 1; done"
```

Comment:
Question: since this loops infinitely the service itself technically never fails. Do we want that? Or do we want to retry a certain amount of times and then have the service fail, so something can report it back up

Reply:
Oh we definitely could improve this and should gather failures somewhere (also related to https://github.com/coreos/ignition/issues/585 ).

It'd also make sense to do at least bounded exponential backoff.

But that said: this is what the installer is doing for the release image today too.  I'm open to doing something more sophisticated but it feels like that can come as a followup - this is a one line change that's going to fix the problem and is "obviously correct".

I offhand can't think of a scenario where we fail to pull the oscontainer that wouldn't have failed elsewhere earlier.  Anyone who does hit some sort of actually permanent scenario would be able to ssh into the node and debug - the same thing that happens with the bootstrap today.